docs/ServiceWorkerPlugin.md

```diff
@@ -0,0 +1,79 @@
+# ServiceWorkerPlugin
+
+Webpack plugin for configuring a ServiceWorker for different PWA development
+scenarios.
+
+## Purpose
+
+This plugin is a wrapper around the [Google Workbox Webpack Plugin](https://developers.google.com/web/tools/workbox/guides/generate-service-worker/).
+It generates a caching ServiceWorker based on assets emitted by Webpack.
+
+This plugin can be configured to run in the following modes:
+
+- *normal development* - ServiceWorker is disabled
+- *service worker debugging* - ServiceWorker and hot-reloading are enabled
+
+## Usage
+
+In `webpack.config.js`:
+
+```js
+const path = require('path');
+const buildpack = require('@magento/pwa-buildpack');
+const ServiceWorkerPlugin = buildpack.Webpack.ServiceWorkerPlugin;
+
+module.exports = async env => {
+    const config = {
+        /* webpack config, i.e. entry, output, etc. */
+        plugins: [
+            /* other plugins */
+            new ServiceWorkerPlugin({
+                env: {
+                    mode: 'development'
```

Comment:
The comment above this code block mentions `normal development`. It should probably specify the string to be used here